### PR TITLE
Refactor Plan trial and setup properties

### PR DESCRIPTION
**Pull request by slavcodev**
Proposal for https://github.com/Rebilly/RebillyAPI/pull/474

### CODE REVIEW

**Comment by slavcodev** [comment]
devs please review

File: spec/definitions/Plans/Plan.yaml [hunk 1]
```diff
@@ -91,25 +91,29 @@ properties:
...
+    allOf:
+      - type: object
+        required:
+          - price
+        properties:
+          price:
+            description: The price of the trial. For a free trial, use `0`.
+            type: number
+            format: double
+      - $ref: "#/definitions/PlanPeriod"
```
Comment (2e3s) [inline spec/definitions/Plans/Plan.yaml]: Why did you move this out?

File: spec/definitions/Plans/PlanPeriod.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+type: object
+required:
```
Comment (2e3s) [inline spec/definitions/Plans/PlanPeriod.yaml]: looks strange

Comment (slavcodev) [reply spec/definitions/Plans/PlanPeriod.yaml]: Yeah, very strange. In my editor it correct: `type`, `required`, etc.

File: spec/definitions/Plans/Plan.yaml (hunk 1 above)
Comment (slavcodev) [reply spec/definitions/Plans/Plan.yaml]: To make specification more explicit. In previous version you can specify trial price w/o to specify trial period. In new version (since we going to change considerably the definition) you can't do that.

Comment (2e3s) [reply spec/definitions/Plans/Plan.yaml]: No, I support the change as I see it, but I meant that why to move the definition in a separate file when it's not reused.

Comment (slavcodev) [reply spec/definitions/Plans/Plan.yaml]: Ahh, it's supposed to be reused for `recuring`.

File: spec/definitions/Plans/PlanPeriod.yaml (hunk 2 above)
Comment (slavcodev) [reply spec/definitions/Plans/PlanPeriod.yaml]: Seems something wrong only with how this diff is shown, the file looks ok
https://github.com/Rebilly/RebillyAPI/blob/8b94b87772ab80150b65df3871a2e9277e3dc9f3/spec/definitions/Plans/PlanPeriod.yaml
